torch/nn/_functions/thnn/sparse.py

```diff
@@ -164,57 +166,60 @@ def forward(self, weight, indices, offsets):
                              " ({}), but got offsets[-1] of {}"
                              .format(indices.size(0), offsets[-1]))
 
-        self._backend = type2backend[type(weight)]
-        self._weight_size = weight.size()
-        self._offset2bag = offsets.new()
+        ctx._backend = type2backend[type(weight)]
+        ctx._weight_size = weight.size()
+        ctx._offset2bag = offsets.new()
 
-        self.save_for_backward(indices)
+        ctx.save_for_backward(indices)
 
         indices = indices.contiguous().view(-1)
         output = weight.new()
-        if self.max_norm is not None:
-            self._renorm(indices, weight)
+
+        if ctx.max_norm is not None:
+            cls._renorm(ctx, indices, weight, max_norm=max_norm, norm_type=norm_type)
```

Comment:
Hmm it's kind of ugly to put `_renorm` in an autograd function, since it's not differentiated 😕 It's not a blocker for this PR in any way, I'll try to think about a nicer solution